src/pr-info.ts

```diff
@@ -404,21 +412,48 @@ configSuspicious["tsconfig.json"] = makeJsonCheckerFromCore(
 // to it, ignoring some keys (JSON Patch paths).  The ignored properties are in most cases checked
 // elsewhere (dtslint), and in some cases they are irrelevant.
 function makeJsonCheckerFromCore(requiredForm: any, ignoredKeys: string[]) {
-    const diffFromReq = (text: string) => {
-        let json: any;
-        try { json = JSON.parse(text); } catch (e) { return "couldn't parse json"; }
-        jsonDiff.applyPatch(json, ignoredKeys.map(path => ({ op: "remove", path })));
-        try { return jsonDiff.compare(requiredForm, json); } catch (e) { return "couldn't diff json" };
-    };
     return (contents: string, oldText?: string) => {
-        const newDiff = diffFromReq(contents);
-        if (typeof newDiff === "string") return newDiff;
+        const diffFromReq = (json: any) => {
+            jsonDiff.applyPatch(json, ignoredKeys.map(path => ({ op: "remove", path })));
```

Comment:
I ***really*** dislike this side-effecting API.  It should really be wrapped in a helper that does the deep cloning.  (This is for the use of `applyPatch` here and below.)

It would also make the code *way* more readable.  For example, I now see an obscure `const suggestion = jsonDiff.deepClone(newJson)` which makes sense only much later in thecode, when it is modified.

In addition to that, I'm not sure that it's correct: there's the line that removes `ignoredKeys` from `json` which would change it??  That should also be simpler if it's done in a functional way.

(Sorry for not commenting about this earlier: the decision to make `applyPatch` modify its input by default seems like such a bad idea that it never occurred to me that this is what it does...)

Reply:
Revised ...

Reply:
Looks much better -- thanks!  (I'll probably go over it again at some point.)